repondeur/zam_repondeur/services/fetch/an/amendements.py

```diff
@@ -505,6 +599,129 @@ def get_organe_abrev(organe_uid: str) -> str:
+        rectif: int,
+        position: Optional[int],
+        id_discussion_commune: Optional[int],
+        id_identique: Optional[int],
+        matricule: str,
+        groupe: str,
+        auteur: str,
+        mission_titre: Optional[str],
+        mission_titre_court: Optional[str],
+        corps: str,
+        expose: str,
+        sort: str,
+    ):
+        self.lecture = lecture
+        self.article = article
+        self.parent_num_raw = parent_num_raw
+        self.rectif = rectif
+        self.position = position
+        self.id_discussion_commune = id_discussion_commune
+        self.id_identique = id_identique
+        self.matricule = matricule
+        self.groupe = groupe
+        self.auteur = auteur
+        self.mission_titre = mission_titre
+        self.mission_titre_court = mission_titre_court
+        self.corps = corps
+        self.expose = expose
+        self.sort = sort
+
+    def _get_parent(self) -> Optional[Amendement]:
+        parent_num, parent_rectif = Amendement.parse_num(self.parent_num_raw)
+        if not parent_num:
+            return None
+        parent: Optional[Amendement]
+        parent, _ = get_one_or_create(
+            Amendement,
+            create_kwargs={"article": self.article, "rectif": parent_rectif},
+            lecture=self.lecture,
+            num=parent_num,
+        )
+        return parent
+
+
+class CreateAmendement(CreateOrUpdateAmendement):
+    def __init__(self, num: int, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.num = num
+
+    def __repr__(self) -> str:
+        return f"<CreateAmendement(num={self.num})>"
+
+    def apply(self) -> FetchResult:
+        amendement = Amendement.create(
+            lecture=self.lecture,
+            article=self.article,
+            parent=self._get_parent(),
+            position=self.position,
+            num=self.num,
+            rectif=self.rectif,
+            id_discussion_commune=self.id_discussion_commune,
+            id_identique=self.id_identique,
+            matricule=self.matricule,
+            groupe=self.groupe,
+            auteur=self.auteur,
+            mission_titre=self.mission_titre,
+            mission_titre_court=self.mission_titre_court,
+            corps=self.corps,
+            expose=self.expose,
+            sort=self.sort,
+        )
+
+        return FetchResult.create(amendements=[amendement], created=1)
+
+
+class UpdateAmendement(CreateOrUpdateAmendement):
+    def __init__(self, amendement: Amendement, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.amendement = amendement
+
+    def __repr__(self) -> str:
+        return f"<UpdateAmendement(num={self.amendement.num})>"
+
+    def apply(self) -> FetchResult:
+        if (
+            self.amendement.location.batch
+            and self.amendement.article.pk != self.article.pk
+        ):
+            BatchUnset.create(amendement=self.amendement, request=None)
+
+        modified = False
```

Comment:
I think we can remove all the `modified` dance here and in `update_*` (now static) methods?